### ISSUE TITLE
Service become unavailable from Russia

### ISSUE THREAD

**Issue by iron-cherep** [outlandishideas/wpackagist#277]
Month ago russian government started blocking Telegram messenger and by the way (maybe temporary) blocked millions of IP addresses including some of Amazon subnetworks. All this time Wpackagist was totally available, but two days ago it become unreachable - both website and composer repository.

As i can see wpackagist.org is hosted at ip 34.246.216.155, which is actually blocked (General Prosecutor’s Office decision "27-31-2018/Ид2971-18" dated the 16 April 2018).

Is there a way to fix it? Not the russian government i mean, but availability of service. Maybe we could do a mirror somehow? This problem totally ruins our normal workflow.

**Comment by joaquimds** [comment]
wpackagist.org is hosted on Amazon Web Services. I could add a second web server in a different AWS region, do you think this would help? Which region should I choose?

**Comment by khromov** [comment]
It is possible to run WPackagist yourself quite easily, there are setup instructions in the README.

For a quick fix, I run a mirror here which is hosted in Frankfurt using DigitalOcean. You can try it by replacing the repository URL in `composer.json` with:
https://wpackagist.wpsv.se/

@joaquimds I think it'd be quite nice actually to have a mirror in `eu-central-1`, `us-west-1` and `ap-south-1` to cover the biggest developer areas and then use Route 53 Geolocation routing policy to route to the closes geographic location. 👍

**Comment by iron-cherep** [comment]
@joaquimds that would be totally awesome if that is possible. Here is list of amazon regions with their availability status: https://rkn.darkbyte.ru/amazon.php

**Comment by joaquimds** [comment]
@iron-cherep As a quick fix, I'm trying to get an elastic IP in one of the reachable blocks on the website you provided.

Can you check that you can do `curl 52.212.28.248 -H "Host: wpackagist.org"`?

**Comment by iron-cherep** [comment]
@joaquimds yep, checked and it works, returns site html as it's supposed to +1

**Comment by joaquimds** [comment]
@iron-cherep OK great. I'll start switching the DNS now, which takes about 3 hours. You can save this time by updating your `hosts` file to point `wpackagist.org` to `52.212.28.248`

**Comment by iron-cherep** [comment]
@joaquimds yeah it works now :rocket: thank you so much for such rapid reaction :1st_place_medal:
